juntagrico/util/management.py

```diff
@@ -64,3 +61,22 @@ def add_subscription_to_member(member, subscription):
             member.old_subscriptions.add(subscription)
         else:
             member.subscription = subscription
+
+
+def replace_subscription_types(subscription, selected_types, _obj=TFSST):
+    """
+    Deletes all types of a subscription and replace them with the types and amounts given by selected_types
+    :param subscription: The Subscription object whose types should be replaced
+    :param selected_types: dictionary of subscription types as key and amount as value
+    :param _obj: for internal use
+    :type _obj: type(TFSST) or type(TSST)
```

Comment:
is _obj really nescessary? isnt if sub not created or wating both and only future otherwise?

Reply:
It is not needed externally. It is just there to reduce code duplication. Could be done with a nested function instead
```
def replace_subscription_types(subscription, selected_types):
    """
    Deletes all types of a subscription and replace them with the types and amounts given by selected_types
    :param subscription: The Subscription object whose types should be replaced
    :param selected_types: dictionary of subscription types as key and amount as value
    :return: None
    """
    def replace(obj):
        """
        :type obj: type(TFSST) or type(TSST)
        """
        obj.objects.filter(subscription=subscription).delete()
        obj.objects.bulk_create(
            itertools.chain(*[[obj(subscription=subscription, type=sub_type)] * amount
                              for sub_type, amount in selected_types.items()])
        )
    # replace future sub types
    replace(TFSST)
    # replace the current sub types as well, if the subscription is not active yet:
    if subscription.state == 'waiting':
        replace(TSST)
```

Reply:
now i understand you called that recursively, i clearly prefer the nested function. on the other hand is the code publication more or less minimal and not so problematic since it is in the same function and next to each other
we have similar patterns in the subscription but we do not use nested functions there so i am not sure about the nested